### PR TITLE
ci: stage golangci-lint, ci: add errcheck linter

### PR DESCRIPTION
**Pull request by acpana**
Branching off from https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1186. 

Commits mostly cherry-picked for errcheck with some fixup.

### CODE REVIEW

File: config/tests/samples/create/samples.go [hunk 1]
```diff
@@ -404,7 +404,9 @@ func replaceResourceNamesWithUniqueIDs(t *testing.T, unstructs []*unstructured.U
 			if err != nil {
 				t.Fatalf("error generating new spec.displayName value for Folder '%v': %v", u.GetName(), err)
 			}
-			unstructured.SetNestedField(newUnstruct.Object, newDisplayName, "spec", "displayName")
+			if err := unstructured.SetNestedField(newUnstruct.Object, newDisplayName, "spec", "displayName"); err != nil {
```
Comment (justinsb) [inline config/tests/samples/create/samples.go:407]: I'm glad to see this, we have some similar changes in the serving path, and so it's good to see test coverage!

File: pkg/controller/lifecyclehandler/handler.go [hunk 2]
```diff
@@ -377,13 +388,13 @@ func setObservedGeneration(resource *k8s.Resource, observedGeneration int64) {
 	resource.Status["observedGeneration"] = observedGeneration
 }
 
-func (r *LifecycleHandler) recordEvent(resource *k8s.Resource, eventtype, reason, message string) error {
+func (r *LifecycleHandler) recordEvent(resource *k8s.Resource, eventtype, reason, message string) {
 	u, err := resource.MarshalAsUnstructured()
 	if err != nil {
-		return err
+		// todo acpana log err
```
Comment (justinsb) [inline pkg/controller/lifecyclehandler/handler.go:394]: Yeah, a sternly worded warning is probably fine here.  I don't love the fact that k8s.Resource is not a wrapper around unstructured.Unstructured or client.Object anyway.

File: pkg/test/main/testmain.go [hunk 3]
```diff
@@ -77,7 +77,9 @@ func TestMainSetupMultipleEnvironments(m *testing.M, testType test.TestType, crd
 		mgr, stop := testcontroller.StartTestManagerInstance(env, testType, whCfgs)
 		stops = append(stops, stop)
 
-		testcontroller.EnsureNamespaceExists(mgr.GetClient(), k8s.SystemNamespace)
+		if err := testcontroller.EnsureNamespaceExists(mgr.GetClient(), k8s.SystemNamespace); err != nil {
+			log.Fatalf("error ensuring namesapce exists: %v", err)
```
Comment (justinsb) [inline pkg/test/main/testmain.go:81]: We should add a common-typo linter at some stage :-)

**Comment by justinsb** [comment]
Thanks for doing this and thanks for breaking it into bite-size chunks for me!

/approve
/lgtm

File: pkg/test/main/testmain.go (hunk 3 above)
Comment (acpana) [reply pkg/test/main/testmain.go:81]: oh it's coming hehe

**Pull request merged**